doc/reference/javascript_cheatsheet.rst

```diff
@@ -276,4 +276,236 @@ using the view in the kanban arch (a specific example is the helpdesk dashboard)
     subjected to the same rules (rng validation, for example).  So, your views still
-    need to have a valid arch field.
\ No newline at end of file
+    need to have a valid arch field.
+
+Promises and asynchronous code
+===============================
+For a very good and complete introduction to promises, please read this excellent article https://github.com/getify/You-Dont-Know-JS/blob/master/async%20%26%20performance/ch3.md
+
+Creating new Promises
+-----------------------
+- turn a constant into a promise
+    There are 2 static functions on Promise that create a resolved or rejected promise based on a constant:
+
+    .. code-block:: javascript
+
+        var p = Promise.resolve({blabla: '1'}); // creates a resolved promise
+        p.then(function (result) {
+            console.log(result); // --> {blabla: '1'};
+        });
+
+
+        var p2 = Promise.reject({error: 'error message'}); // creates a rejected promise
+        p2.catch(function (reason) {
+            console.log(reason); // --> {error: 'error message');
+        });
+
+
+    .. note:: Note that even if the promises are created already resolved or rejected, the `then` or `catch` handlers will still be called asynchronously.
+
+
+- based on an already asynchronous code
+    Suppose that in a function you must do a rpc, and when it is completed set the result on this.
+    The `this._rpc` is a function that returns a `Promise`.
+
+    .. code-block:: javascript
+
+        function callRpc() {
+            var self = this;
+            return this._rpc(...).then(function (result) {
+                self.myValueFromRpc = result;
+            });
+        }
+
+- for callback based function
+    Suppose that you were using a function `this.close` that takes as parameter a callback that is called when the closing is finished.
+    Now suppose that you are doing that in a method that must send a promise that is resolved when the closing is finished.
+
+    .. code-block:: javascript
+        :linenos:
+
+        function waitForClose() {
+            var self = this;
+            return new Promise (function(resolve, reject) {
+                self.close(resolve);
+            });
+        }
+
+    * line 2: we save the `this` into a variable so that in an inner function, we can access the scope of our component
+    * line 3: we create and return a new promise. The constructor of a promise takes a function as parameter. This function itself has 2 parameters that we called here `resolve` and `reject`
+        - `resolve` is a function that, when called, puts the promise in the resolved state.
+        - `reject` is a function that, when called, puts the promise in the rejected state. We do not use reject here and it can be omitted.
+    * line 4: we are calling the function close on our object. It takes a function as parameter (the callback) and it happens that resolve is already a function, so we can pass it directly. To be clearer, we could have written:
```

Comment:
I would inline the comments, I find it very hard to read otherwise, but that might just be me.